File: VB/More/Watchdog/WriteWatchdogConfig/WriteWatchdogConfig.vb
'------------------------------------------------------------------------------
' WriteWatchdogConfig.vb
'
' Demonstrates how to configure the Watchdog on a LabJack.
'
' support@labjack.com
'------------------------------------------------------------------------------
Option Explicit On

Imports LabJack


Module WriteWatchdogConfig

    Sub Main()
        Dim handle As Integer
        Dim numFrames As Integer
        Dim aNames() As String
        Dim aValues() As Double
        Dim errAddr As Integer = -1

        Try
            ' Open first found LabJack
            LJM.OpenS("ANY", "ANY", "ANY", handle)  ' Any device, Any connection, Any identifier
            'LJM.OpenS("T7", "ANY", "ANY", handle)  ' T7 device, Any connection, Any identifier
            'LJM.OpenS("T4", "ANY", "ANY", handle)  ' T4 device, Any connection, Any identifier
            'LJM.Open(LJM.CONSTANTS.dtANY, LJM.CONSTANTS.ctANY, "ANY", handle)  ' Any device, Any connection, Any identifier

            displayHandleInfo(handle)

            ' Setup and call eWriteNames to configure the Watchdog on a
            ' LabJack. Disable the Watchdog first before any other
            ' configuration.
            numFrames = 16
            ReDim aNames(numFrames - 1)
            aNames(0) = "WATCHDOG_ENABLE_DEFAULT"
            aNames(1) = "WATCHDOG_ADVANCED_DEFAULT"
            aNames(2) = "WATCHDOG_TIMEOUT_S_DEFAULT"
            aNames(3) = "WATCHDOG_STARTUP_DELAY_S_DEFAULT"
            aNames(4) = "WATCHDOG_STRICT_ENABLE_DEFAULT"
            aNames(5) = "WATCHDOG_STRICT_KEY_DEFAULT"
            aNames(6) = "WATCHDOG_RESET_ENABLE_DEFAULT"
            aNames(7) = "WATCHDOG_DIO_ENABLE_DEFAULT"
            aNames(8) = "WATCHDOG_DIO_STATE_DEFAULT"
            aNames(9) = "WATCHDOG_DIO_DIRECTION_DEFAULT"
            aNames(10) = "WATCHDOG_DIO_INHIBIT_DEFAULT"
            aNames(11) = "WATCHDOG_DAC0_ENABLE_DEFAULT"
            aNames(12) = "WATCHDOG_DAC0_DEFAULT"
            aNames(13) = "WATCHDOG_DAC1_ENABLE_DEFAULT"
            aNames(14) = "WATCHDOG_DAC1_DEFAULT"
            aNames(15) = "WATCHDOG_ENABLE_DEFAULT"
            ReDim aValues(numFrames - 1)
            aValues(0) = 0
            aValues(1) = 0
            aValues(2) = 20
            aValues(3) = 0
            aValues(4) = 0
            aValues(5) = 0
            aValues(6) = 1
            aValues(7) = 0
            aValues(8) = 0
            aValues(9) = 0
            aValues(10) = 0
            aValues(11) = 0
            aValues(12) = 0
            aValues(13) = 0
            aValues(14) = 0
            aValues(15) = 0  ' Set WATCHDOG_ENABLE_DEFAULT to 1 to enable
            LJM.eWriteNames(handle, numFrames, aNames, aValues, errAddr)

            Console.WriteLine("")
            Console.WriteLine("Set Watchdog configuration:")
            For i = 0 To numFrames - 1
                Console.WriteLine("    " & aNames(i) & " : " & aValues(i))
            Next
        Catch ljme As LJM.LJMException
            showErrorMessage(ljme)
        End Try

        LJM.CloseAll()  ' Close all handles

        Console.WriteLine("")
        Console.WriteLine("Done.")
        Console.WriteLine("Press the enter key to exit.")
        Console.ReadLine()  ' Pause for user
    End Sub

End Module

File: C#/More/Watchdog/WriteWatchdogConfig/WriteWatchdogConfig.cs
//-----------------------------------------------------------------------------
// WriteWatchdogConfig.cs
//
// Demonstrates how to configure the Watchdog on a LabJack.
//
// support@labjack.com
//
// Relevant Documentation:
//
// LJM Library:
//     LJM Library Installer:
//         https://labjack.com/support/software/installers/ljm
//     LJM Users Guide:
//         https://labjack.com/support/software/api/ljm
//     Opening and Closing:
//         https://labjack.com/support/software/api/ljm/function-reference/opening-and-closing
//     eWriteNames:
//         https://labjack.com/support/software/api/ljm/function-reference/ljmewritenames
//
// T-Series and I/O:
//     Modbus Map:
//         https://labjack.com/support/software/api/modbus/modbus-map
//     Watchdog:
//         https://labjack.com/support/datasheets/t-series/watchdog
//-----------------------------------------------------------------------------
using System;
using LabJack;


namespace WriteWatchdogConfig
{
    class WriteWatchdogConfig
    {
        static void Main(string[] args)
        {
            WriteWatchdogConfig wwc = new WriteWatchdogConfig();
            wwc.performActions();
        }

        public void showErrorMessage(LJM.LJMException e)
        {
            Console.Out.WriteLine("LJMException: " + e.ToString());
            Console.Out.WriteLine(e.StackTrace);
        }

        public void performActions()
        {
            int handle = 0;
            int devType = 0;
            int conType = 0;
            int serNum = 0;
            int ipAddr = 0;
            int port = 0;
            int maxBytesPerMB = 0;
            string ipAddrStr = "";

            try
            {
                //Open first found LabJack
                LJM.OpenS("ANY", "ANY", "ANY", ref handle);  // Any device, Any connection, Any identifier
                //LJM.OpenS("T8", "ANY", "ANY", ref handle);  // T8 device, Any connection, Any identifier
                //LJM.OpenS("T7", "ANY", "ANY", ref handle);  // T7 device, Any connection, Any identifier
                //LJM.OpenS("T4", "ANY", "ANY", ref handle);  // T4 device, Any connection, Any identifier
                //LJM.Open(LJM.CONSTANTS.dtANY, LJM.CONSTANTS.ctANY, "ANY", ref handle);  // Any device, Any connection, Any identifier

                LJM.GetHandleInfo(handle, ref devType, ref conType, ref serNum, ref ipAddr, ref port, ref maxBytesPerMB);
                LJM.NumberToIP(ipAddr, ref ipAddrStr);
                Console.WriteLine("Opened a LabJack with Device type: " + devType + ", Connection type: " + conType + ",");
                Console.WriteLine("Serial number: " + serNum + ", IP address: " + ipAddrStr + ", Port: " + port + ",");
                Console.WriteLine("Max bytes per MB: " + maxBytesPerMB);

                //Setup and call eWriteNames to configure the Watchdog on a
                //LabJack. Disable the Watchdog first before any other
                //configuration.
                string[] aNames = new string[] {
                    "WATCHDOG_ENABLE_DEFAULT", "WATCHDOG_ADVANCED_DEFAULT",
                    "WATCHDOG_TIMEOUT_S_DEFAULT", "WATCHDOG_STARTUP_DELAY_S_DEFAULT",
                    "WATCHDOG_STRICT_ENABLE_DEFAULT", "WATCHDOG_STRICT_KEY_DEFAULT",
                    "WATCHDOG_RESET_ENABLE_DEFAULT", "WATCHDOG_DIO_ENABLE_DEFAULT",
                    "WATCHDOG_DIO_STATE_DEFAULT", "WATCHDOG_DIO_DIRECTION_DEFAULT",
                    "WATCHDOG_DIO_INHIBIT_DEFAULT", "WATCHDOG_DAC0_ENABLE_DEFAULT",
                    "WATCHDOG_DAC0_DEFAULT", "WATCHDOG_DAC1_ENABLE_DEFAULT",
                    "WATCHDOG_DAC1_DEFAULT", "WATCHDOG_ENABLE_DEFAULT"};
                double[] aValues = new double[] {
                    0, 0,
                    20, 0,  //Timeout set to 20 seconds
                    0, 0,
                    1, 0,  //Reset the device on timeout
                    0, 0,
                    0, 0,
                    0, 0,
                    0, 0};  //Change the last WATCHDOG_ENABLE_DEFAULT to 1 to enable
                int numFrames = aNames.Length;
                int errAddr = -1;
                LJM.eWriteNames(handle, numFrames, aNames, aValues, ref errAddr);

                Console.WriteLine("\nSet Watchdog configuration:");
                for(int i = 0; i < numFrames; i++)
                    Console.WriteLine("    " + aNames[i] +" : " + aValues[i]);

            }
            catch (LJM.LJMException e)
            {
                showErrorMessage(e);
            }

            LJM.CloseAll();  //Close all handles

            Console.WriteLine("\nDone.\nPress the enter key to exit.");
            Console.ReadLine();  //Pause for user
        }
    }
}
